### ISSUE TITLE
Prefer C++ "enum class" to "enum" 

### ISSUE THREAD

**Issue by hjmjohnson** [InsightSoftwareConsortium/ITK#1049]
### Description

Prefer C++ "enum class" to "enum" 

https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum

### Impact analysis

Places where types are implicitly converted, may now need explicit static_casts.

### Expected behavior

Other than being explicit with type conversions, this should provide more careful conversion and strict usage of enumerations.

### Actual behavior

```c++
    typedef enum {
      MUSTFLUSH = 0,
      FATAL,
      CRITICAL,
      WARNING,
      INFO,
      DEBUG,
      NOTSET
      } PriorityLevelType;
```
to
```c++
 enum class PriorityLevelEnum: uint8_t
   {
      MUSTFLUSH = 0,
      FATAL,
      CRITICAL,
      WARNING,
      INFO,
      DEBUG,
      NOTSET
  };
```

**Comment by hjmjohnson** [comment]
FYI:  Still working on this.  It is a huge amount of (nearly) mindless work.  I'll be glad when it is finished.

**Comment by dzenanz** [comment]
Was this fixed by #1471?

**Comment by hjmjohnson** [comment]
FIXED in #1471
